packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -449,17 +464,33 @@ extern dec useSystemTextJsonConverter(target: Model, scope?: valueof string);
  *   blobName: string;
  * }
  *
- * @@clientInitialization(MyService, MyServiceClientOptions)
- * // The generated client will have `blobName` on it. We will also
- * // elevate the existing `blobName` parameter to the client level.
+ * @@clientInitialization(MyService, {parameters: MyServiceClientOptions})
+ * // The generated client will have `blobName` on its initialization method. We will also
+ * // elevate the existing `blobName` parameter from method level to client level.
  * ```
  */
 extern dec clientInitialization(
   target: Namespace | Interface,
-  options: Model,
+  options: ClientInitializationOptions,
   scope?: valueof string
 );
 
+/**
+ * Client initialization customization options.
+ */
+model ClientInitializationOptions {
+  /**
+   * Redefine the client initialization parameters you would like to add to the client.
+   * By default, we apply endpoint, credential, and api-version parameters. If you specify parameters model, we will append the properties of the model to the parameters list of the client initialization.
+   */
+  parameters?: Model;
+
+  /**
+   * Determines how the client could be initialized. Use `InitializedBy` enum to set the value. The value could be `InitializedBy.individually`, `InitializedBy.parent` or `InitializedBy.individually | InitializedBy.parent`.
+   */
+  initializedBy?: EnumMember | Union;
```

Comment:
Why is this optional? What is the default when this is not defined? Will this assume `individually` or `parent` or both? We should update the docs here to clarify.

Reply:
This part is in decorator, which mean the dev does not have to specify it.

I think Chenjie have this be "required" in TCGC API, so TCGC always gives a value.